map.js

```diff
@@ -195,143 +183,97 @@ function drawMapOverlay(){
         this.minY = y;
-        this.maxX = x+width;
         this.maxY = y+height;
         this.name = text;
-        this.draw = function(ctx){
-            let width = this.width();
-            let height = this.height();
-            ctx.beginPath();
-            if(currentOverlay == this.name){
-                ctx.fillStyle = "#ccc";
-            }
-            else{
-                ctx.fillStyle = "#E5D9B9";
-            }
-            ctx.fillRect(this.minX, this.minY, width, height);
-
-            //and now text
-            ctx.beginPath();
-            ctx.fillStyle = "#000000";
-            ctx.textAlign = "center";
-            ctx.textBaseline = "alphabetic";
-            ctx.font = "16px Arial"
-            ctx.fillText(this.name, this.minX + (width/2), this.minY + 16);
-        }
+        this.recalculateBoundingBox();
+        
     }
     MapButton.prototype = Object.create(MapObject.prototype);
+    MapButton.prototype.draw = function(ctx){
+        let width = this.width();
+        let height = this.height();
+        ctx.beginPath();
+        if(currentOverlay == this.name){
+            ctx.fillStyle = "#ccc";
+        }
+        else{
+            ctx.fillStyle = "#E5D9B9";
+        }
+        ctx.fillRect(this.minX, this.minY, width, height);
 
-    topbar = new MapObject();
-    topbar.buttons = [];
-    topbar.minX = 0;
-    topbar.minY = 0;
-    topbar.maxX = viewport.clientWidth;
-    topbar.maxY = 32;
+        //and now text
+        ctx.beginPath();
+        ctx.fillStyle = "#000000";
+        ctx.textAlign = "center";
+        ctx.textBaseline = "alphabetic";
+        ctx.font = "16px Arial"
+        ctx.fillText(this.name, this.minX + (width/2), this.minY + 16);
+    }
+    MapButton.prototype.recalculateBoundingBox = function(){
+        this.minX = 8 + (map_topbar.width()/3)*this.ordinal;
+        this.maxX = this.minX + (map_topbar.width() - 16)/3;
+    }
+
+    map_topbar = new MapObject();
+    map_topbar.buttons = [];
+    map_topbar.minX = 0;
+    map_topbar.minY = 0;
+    map_topbar.maxX = viewport.clientWidth;
+    map_topbar.maxY = 32;
 
-    topbar.buttons.push(new MapButton(8,6, topbar.width()/3, 20,"Locations"));
-    topbar.buttons.push(new MapButton(topbar.width()/3,6, topbar.width()/3, 20,"NirnRoute"));
-    topbar.buttons.push(new MapButton(topbar.width()/3*2, 6, topbar.width()/3 - 8, 20, "Exploration"));
+    map_topbar.buttons.push(new MapButton(0, 6, 20, "Locations"));
+    map_topbar.buttons.push(new MapButton(1, 6, 20, "NirnRoute"));
+    map_topbar.buttons.push(new MapButton(2, 6, 20, "Exploration"));
 
-    topbar.draw = function(ctx){
+    map_topbar.draw = function(ctx){
         let wX = viewport.clientWidth;
 
         //update our width here for hit detection
-        this.maxX = wX;
+        if(this.maxX != wX){
+            this.maxX = wX;
+            for(const btn of this.buttons){
+                btn.recalculateBoundingBox();
+            }
+        }
```

Comment:
This recalculates button width on draw, not on window resize, but this PR is large enough already